### DISCUSSION TITLE
pickup&delivery - how to maximize the use of existing vehicles

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#3488] [Routing (and legacy CP) questions] [answered]
In a pickup&delivery setting, how to maximize the use of existing vehicle on the road.   

in my model, I have set the depot as dummy node, with it's distance to all other nodes 0. this means, the model will be encouraged to generate a vehicle at the start of a route, whether this route is for visiting a single node or multiple nodes (i.e. consolidation). however, I want the model to assign task to the existing vehicle rather than new vehicle. i.e. after an existing vehicle unload it's goods, the model immediately assign this vehicle to drive to another place to load another goods, rather than generate a new vehicle at that place. 

For sure using existing vehicle might increase the cost as it's associated with additional empty run from it's current unload place to the new pickup place, while the new one will start right at the new place, but in reality this is not practical as no one can be sure if a new vehicle can be found at the pickup place. in this case, we want to maximize the use of the existing vehicle (most are self-owned) to do round trip or connected trips. Then, after the current fleet is fully exploited, we can ask the model to add some new vehicles (in reality this means outsourcing).

**Comment by Mizux** [comment, marked as answer]
Did you try to add/play with Fixed Vehicle cost ?
ref: https://github.com/google/or-tools/blob/82750ac12f1ee5354e1c7869894d9af3508778f2/ortools/constraint_solver/routing.h#L1038-L1046

**Reply by lperron** [reply]
Please indicate when you cross post from the mailing list. Now, you have 2 persons that has answered the same idea.
